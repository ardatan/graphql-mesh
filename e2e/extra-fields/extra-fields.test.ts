import { createTenv } from '@e2e/tenv';

const { serve, compose, service } = createTenv(__dirname);

it('works', async () => {
  const { output } = await compose({
    services: [await service('foo'), await service('bar')],
    output: 'graphql',
  });
  const { execute } = await serve({ supergraph: output });
  await expect(
    execute({
      query: /* GraphQL */ `
        query FooBarFoo {
          foo {
            id
            bar {
              id
              foo {
                id
              }
            }
          }
        }
      `,
    }),
  ).resolves.toMatchSnapshot();
});
