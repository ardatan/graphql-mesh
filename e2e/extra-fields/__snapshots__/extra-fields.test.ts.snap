// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`works 1`] = `
{
  "data": {
    "foo": {
      "bar": {
        "foo": {
          "id": "1",
        },
        "id": "1",
      },
      "id": "1",
    },
  },
}
`;
